File: Domain/Entities/Main/Inheritances/MainEntitiesBase.cs


using System.Collections.Generic;
using Domain.Entities.Main.Inheritances.Enums;
using Domain.Entities.Outsourced;
using Domain.Entities.Shared;
using Domain.Entities.StkProduct;

namespace Domain.Entities.Main.Inheritances
{
    public class MainEntitiesBase:RootBase
    {
        public string Name { get; set; }
        public string Responsible { get; set; }
        public string CNPJ { get; set; }
        public EntityTypeEnum EntityType { get; set; }
        public string Description { get; set; }
        public string BusinessLine { get; set; }
        public Address Address { get; set; }
        public Contact Contact { get; set; }
        public PhysicallyMovingCosts PhysicallyMovingCosts { get; set; }
        public List<Destiny> CollectDeliverDestinies { get; set; }
        public List<BillingFrom> BillingFromCollectsDelivers { get; set; }
        public List<Quantity> ProductsQuantities { get; set; }
    }


}

File: Views/src/shared/entities-dtos/main/inheritances/main-entities-base.ts

import { AddressDto } from "src/shared/entities-dtos/address-dto"
import { CompanyDto } from "src/shared/entities-dtos/company-dto"
import { ContactDto } from "src/shared/entities-dtos/contact-dto";
import { PhysicallyMovingCostsDto } from "./physically-moving-costs";
import { QuantityDto } from "src/components/product/dtos/quantity-dto";
import { EntityTypeEnumDto } from "./enum/entity-type.enum-dto";

export class MainEntitiesBase {
  id: number;
  companyId: number;
  company: CompanyDto;
  name: string;
  responsible: string;
  cnpj: string;
  entityType:EntityTypeEnumDto
  registered: Date;
  description: string;
  businessLine:string;
  address: AddressDto;
  contact: ContactDto;
  physicallyMovingCosts: PhysicallyMovingCostsDto;
  // collectDeliverDestinies: DestinyDto[];
  // billingFromCollectsDelivers: BillingFromDto[];
  productsQuantities: QuantityDto[];
}

